apis/python/src/tiledbsoma/_read_iters.py

```diff
@@ -66,8 +67,47 @@
 class TableReadIter(somacore.ReadIter[pa.Table]):
     """Iterator over `Arrow Table <https://arrow.apache.org/docs/python/generated/pyarrow.Table.html>`_ elements"""
 
-    def __init__(self, sr: clib.SOMAArray):
-        self._reader = _arrow_table_reader(sr)
+    def __init__(
+        self,
+        array: SOMAArray,
+        coords: Union[
+            options.SparseDFCoords, options.SparseNDCoords, options.DenseNDCoords
+        ],
+        column_names: Optional[Sequence[str]],
+        result_order: clib.ResultOrder,
+        value_filter: Optional[str],
+        platform_config: Optional[options.PlatformConfig],
+    ):
+        """Initalizes a new TableReadIter for SOMAArrays.
+
+        Args:
+            array (SOMAArray):
+                The NDArray, DataFrame, or SpatialDataFrame being read.
+
+            coords (Union[
+                options.SparseDFCoords, options.SparseNDCoords, options.DenseNDCoords
+            ]):
+                for each index dimension, which rows to read.
+                ``()`` means no constraint -- all IDs.
+
+            column_names (Optional[Sequence[str]]):
+                The named columns to read and return.
+                ``None`` means no constraint -- all column names.
+
+            result_order (clib.ResultOrder):
+                Order of read results.
```

Comment:
I was going to ask you to list out the valid options here -- unless ` clib.ResultOrder` is clickable in readthedocs, in which case never mind